### DISCUSSION TITLE
texImage2D: width or height out of range when I use TextLayer to render characters.

### DISCUSSION THREAD

**Discussion by iwangyuchun** [visgl/deck.gl#6540] [Q&A]
I want to render too many characters, and the current texture size is not enough. Is there a way to expand the size of the texture TextLayer uses.

**Comment by Pessimistress** [comment]
Texture size is limited by https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/WebGL_best_practices#understand_system_limits

You can use a smaller [fontSettings.fontSize](https://deck.gl/docs/api-reference/layers/text-layer#fontsettings).

**Reply by iwangyuchun** [reply]
I got it, Thank you very much.
